src/EnergyPlus/Coils/CoilCoolingDX.cc

```diff
@@ -575,15 +575,15 @@ void CoilCoolingDX::oneTimeInit(EnergyPlusData &state)
 int CoilCoolingDX::getNumModes()
 {
     int numModes = 1;
-    if (this->performance.hasAlternateMode) {
+    if (this->performance.coilMode != HVAC::CoilMode::Normal) {
         numModes++;
     }
     return numModes;
 }
 
-int CoilCoolingDX::getOpModeCapFTIndex(bool const useAlternateMode)
+int CoilCoolingDX::getOpModeCapFTIndex(HVAC::CoilMode const mode)
```

Comment:
This is an example of brittle code. Using the `CoilMode` as a `bool` relies on the fact that `Normal` is zero. Don't obfuscate. If you are trying to distinguish between `Normal` mode and the other modes just make that comparison directly.